CONTRIBUTING.md

```diff
@@ -0,0 +1,49 @@
+<!--
+{% comment %}
+Copyright 2020 Graz University of Technology
+
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% end comment %}
+-->
+
+# Contributing to SystemDS
+
+Thanks for taking the time to contribute to SystemDS!
+
+The following are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+
+___
+
+## Code Style
+
+Before contributing a pull request, we highly suggest applying a code formatter to the written code.
+
+We have provided at profile for java located in [Codestyle File ./docs/CodeStyle.eclipse.xml](./docs/CodeStyle_eclipse.xml). This can be loaded in most editors e.g.:
+
+- [Eclipse](https://stackoverflow.com/questions/10432538/eclipse-import-conf-xml-files#10433986)
+- [IntelliJ](https://imagej.net/Eclipse_code_style_profiles_and_IntelliJ)
+- [Visual Studio Code](https://stackoverflow.com/questions/46030629/need-to-import-eclipse-java-formatter-profile-in-visual-studio-code)
+
+## License
+
+Each file in the project has a header license, when adding files remember to add an appropriate comment for that filetype containing the license. One Example is this file: [RAW Link](https://raw.githubusercontent.com/tugraz-isds/systemds/master/CONTRIBUTING.md).
+
+The specific type of commenting `\** *\` or `# ...` varies depending on filetype, but the content is the same except for year and "Modifications".
+A rule of thumb is that the License should be updated if more than 5 lines of code changed in the file.
+
```

Comment:
Not sure if the copyright, including the date, should change every time one adds code to a file. That will make it hard to easily find when a file was created (without going through the history), and blow up every commit a bit by all the header modifications. Like to hear other's opinions.
Fine with the other mentioned points.
Thanks for writing up the guideline. 🥇

Reply:
This looks fine by me, except when adding a new file the license is default Apache 2.0 with our Copyright. Only when modifying existing files that have the ASF license, we put our Modification Copyright on top.

Reply:
One more comment: why are we linking to the CONTRIBUTING.md itself - am I missing something?

Reply:
The link to itself is because if ppl are reading this file on GitHub, it does not show the license, except if it is in RAW mode, which the link is referring to.

I have now added the licenses in clear text.